File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = new Map();
    
    for(const index in nums){
        if(map.has(target-nums[index])){
            return [map.get(target-nums[index]), index];
        }else{
            map.set(nums[index], index);
        }
    }
};

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> um;
        vector<int> ans;
        int n = nums.size();
        for(int i=0; i<n; i++){
            int val = target - nums[i];
            if(um.find(val) != um.end()){
                 ans.push_back(um[val]);
                ans.push_back(i);
            }else{
               um[nums[i]] = i;
            }
        }
        return ans;
    }
};